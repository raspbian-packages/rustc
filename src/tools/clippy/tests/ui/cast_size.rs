#[warn(cast_precision_loss, cast_possible_truncation, cast_sign_loss, cast_possible_wrap, cast_lossless)]
#[allow(no_effect, unnecessary_operation)]
fn main() {
    // Casting from *size
    1isize as i8;
    1isize as f64;
    1usize as f64;
    1isize as f32;
    1usize as f32;
    1isize as i32;
    1isize as u32;
    1usize as u32;
    1usize as i32;
    // Casting to *size
    1i64 as isize;
    1i64 as usize;
    1u64 as isize;
    1u64 as usize;
    1u32 as isize;
    1u32 as usize; // Should not trigger any lint
    1i32 as isize; // Neither should this
    1i32 as usize;
}
