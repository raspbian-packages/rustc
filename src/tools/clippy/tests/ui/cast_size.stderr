error: casting isize to i8 may truncate the value
 --> $DIR/cast_size.rs:5:5
  |
5 |     1isize as i8;
  |     ^^^^^^^^^^^^
  |
  = note: `-D cast-possible-truncation` implied by `-D warnings`

error: casting isize to f64 causes a loss of precision on targets with 64-bit wide pointers (isize is 64 bits wide, but f64's mantissa is only 52 bits wide)
 --> $DIR/cast_size.rs:6:5
  |
6 |     1isize as f64;
  |     ^^^^^^^^^^^^^
  |
  = note: `-D cast-precision-loss` implied by `-D warnings`

error: casting usize to f64 causes a loss of precision on targets with 64-bit wide pointers (usize is 64 bits wide, but f64's mantissa is only 52 bits wide)
 --> $DIR/cast_size.rs:7:5
  |
7 |     1usize as f64;
  |     ^^^^^^^^^^^^^

error: casting isize to f32 causes a loss of precision (isize is 32 or 64 bits wide, but f32's mantissa is only 23 bits wide)
 --> $DIR/cast_size.rs:8:5
  |
8 |     1isize as f32;
  |     ^^^^^^^^^^^^^

error: casting usize to f32 causes a loss of precision (usize is 32 or 64 bits wide, but f32's mantissa is only 23 bits wide)
 --> $DIR/cast_size.rs:9:5
  |
9 |     1usize as f32;
  |     ^^^^^^^^^^^^^

error: casting isize to i32 may truncate the value on targets with 64-bit wide pointers
  --> $DIR/cast_size.rs:10:5
   |
10 |     1isize as i32;
   |     ^^^^^^^^^^^^^

error: casting isize to u32 may lose the sign of the value
  --> $DIR/cast_size.rs:11:5
   |
11 |     1isize as u32;
   |     ^^^^^^^^^^^^^
   |
   = note: `-D cast-sign-loss` implied by `-D warnings`

error: casting isize to u32 may truncate the value on targets with 64-bit wide pointers
  --> $DIR/cast_size.rs:11:5
   |
11 |     1isize as u32;
   |     ^^^^^^^^^^^^^

error: casting usize to u32 may truncate the value on targets with 64-bit wide pointers
  --> $DIR/cast_size.rs:12:5
   |
12 |     1usize as u32;
   |     ^^^^^^^^^^^^^

error: casting usize to i32 may truncate the value on targets with 64-bit wide pointers
  --> $DIR/cast_size.rs:13:5
   |
13 |     1usize as i32;
   |     ^^^^^^^^^^^^^

error: casting usize to i32 may wrap around the value on targets with 32-bit wide pointers
  --> $DIR/cast_size.rs:13:5
   |
13 |     1usize as i32;
   |     ^^^^^^^^^^^^^
   |
   = note: `-D cast-possible-wrap` implied by `-D warnings`

error: casting i64 to isize may truncate the value on targets with 32-bit wide pointers
  --> $DIR/cast_size.rs:15:5
   |
15 |     1i64 as isize;
   |     ^^^^^^^^^^^^^

error: casting i64 to usize may lose the sign of the value
  --> $DIR/cast_size.rs:16:5
   |
16 |     1i64 as usize;
   |     ^^^^^^^^^^^^^

error: casting i64 to usize may truncate the value on targets with 32-bit wide pointers
  --> $DIR/cast_size.rs:16:5
   |
16 |     1i64 as usize;
   |     ^^^^^^^^^^^^^

error: casting u64 to isize may truncate the value on targets with 32-bit wide pointers
  --> $DIR/cast_size.rs:17:5
   |
17 |     1u64 as isize;
   |     ^^^^^^^^^^^^^

error: casting u64 to isize may wrap around the value on targets with 64-bit wide pointers
  --> $DIR/cast_size.rs:17:5
   |
17 |     1u64 as isize;
   |     ^^^^^^^^^^^^^

error: casting u64 to usize may truncate the value on targets with 32-bit wide pointers
  --> $DIR/cast_size.rs:18:5
   |
18 |     1u64 as usize;
   |     ^^^^^^^^^^^^^

error: casting u32 to isize may wrap around the value on targets with 32-bit wide pointers
  --> $DIR/cast_size.rs:19:5
   |
19 |     1u32 as isize;
   |     ^^^^^^^^^^^^^

error: casting i32 to usize may lose the sign of the value
  --> $DIR/cast_size.rs:22:5
   |
22 |     1i32 as usize;
   |     ^^^^^^^^^^^^^

error: aborting due to 19 previous errors

